src/main/java/com/develhope/spring/model/user/Rental.java

```diff
@@ -0,0 +1,47 @@
+package com.develhope.spring.model.user;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+public class Rental {
+    private LocalDate startDate;
+    private LocalDate endDate;
+    private BigDecimal dailyCost;
+    private BigDecimal totalCost;
+    private boolean paid;
+
+    //it was decided to use vehicle (object) but instead of using
+    // Vehicle as object is created private int vehicleId;
+    // passed to the constructor
+    // handled the same way in the Order class of User
+    private int vehicleId;
+
+
+    public Rental(LocalDate startDate, LocalDate endDate, BigDecimal dailyCost, int vehicleId) {
+        this.startDate = startDate;
+        this.endDate = endDate;
+        this.dailyCost = dailyCost;
+        this.vehicleId = vehicleId;
+        this.calculateTotalCost();
+        this.paid = false;
+    }
+
+    private void calculateTotalCost() {
+        long rentalDays = startDate.until(endDate).getDays();
+        this.totalCost = dailyCost.multiply(BigDecimal.valueOf(rentalDays));
+    }
+
+    public void markAsPaid() {
```

Comment:
mancano i setter, difatti questo markAsPaid dovrebbe usare il setPaid per cambiare eventualmente da true a false.

Reply:
up

Reply:
credo si tratti di una questione di stile, sei abituato a pensare a getter e setter ma se parliamo di un booleano che ha solo due valori possibili allora abbiamo l'approccio di cui sopra.

dove usi

this.paid = true;

senza passare per il get e leggere lo stato

sto cercando di capire, ho trovato una risorsa che può essere utile 

https://stackoverflow.com/questions/4851337/setter-for-a-boolean-variable-named-like-isactive

Reply:
Se non è possibile passare da "paid = true" a "paid = false" ci può stare avere un metodo che ti segna l'ordine come pagato ed a cui non è possibile tornare indietro, ovviamente considerando eventuali aggiornamenti del database di conseguenza.

Reply:
va bene, quindi seguendo ciò che hai detto 
avrò paid impostato di default a false (che per il momento è stato commentato per provare a soddisfare la richiesta)

        this.paid = false;
        
        poi come di seguito avremo il metodo 
        
            public boolean isPaid() {
        return paid;
    }

che la mette a true.

non sarà possibile tornare indietro perchè non abbiamo metodi per tornare  a false.

Idealmente mi sembra logico tenere a false di base e poi aggiornare a true piuttosto che il contrario.